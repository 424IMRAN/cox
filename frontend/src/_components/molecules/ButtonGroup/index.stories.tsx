import { Story, Meta } from '@storybook/react'
import { ThemeProvider } from '@material-ui/core/styles'
import ButtonGroup from '.'
import baseTheme from '../../../core-utils/theme'

export default {
  title: 'molecules/ButtonGroup',
  component: ButtonGroup,
  argTypes: {},
} as Meta

const Template: Story = ({ ...args }) => (
  <ThemeProvider theme={baseTheme}>
    <ButtonGroup labels={[]} activeIndex={0} {...args} />
  </ThemeProvider>
)

export const DefaultButtonGroup = Template.bind({})
DefaultButtonGroup.args = {
  labels: ['A - Z', 'Z - A'],
  activeIndex: 0,
}
