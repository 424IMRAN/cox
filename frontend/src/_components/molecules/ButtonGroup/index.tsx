/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react'
import { makeStyles, Theme } from '@material-ui/core/styles'
import clsx from 'clsx'
import Typography from '../../atoms/Typography'
import { COLORS } from '../../../core-utils/theme'

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    backgroundColor: COLORS.TRANSPARENT_GREY_300,
    maxWidth: 'max-content',
    borderRadius: theme.spacing(3),
    padding: theme.spacing(0.5),
    '&&:after': {},
  },
  typogrpahy: {
    minWidth: '68px',
    textAlign: 'center',
  },
  activeTypoGraphyDiv: {
    backgroundColor: COLORS.TERTIARY_GREY_100,
  },
  typographyDiv: {
    padding: `${theme.spacing(1.5)}px ${theme.spacing(4)}px`,
    borderRadius: `${theme.spacing(3)}px`,
  },
  activeTypoGraphy: {
    color: COLORS.TEXT_900,
  },
}))

type ButtonGroupProps = {
  labels?: Array<string>
  activeIndex?: number
  setActive?: (idx: number) => void
}

const ButtonGroup = ({ labels, activeIndex, setActive }: ButtonGroupProps) => {
  const classes = useStyles()
  const handleOnClick = (idx: number) => () => {
    setActive?.(idx)
  }
  return (
    <div className={classes.root} style={{ cursor: 'pointer' }}>
      {labels?.map((label, idx) => (
        <div
          className={clsx(
            classes.typographyDiv,
            activeIndex === idx && classes.activeTypoGraphyDiv
          )}
          onClick={handleOnClick(idx)}
          key={label}
        >
          <Typography
            variant="body2"
            className={clsx(
              classes.typogrpahy,
              activeIndex === idx && classes.activeTypoGraphy
            )}
          >
            {label}
          </Typography>
        </div>
      ))}
    </div>
  )
}

ButtonGroup.propTypes = {}
ButtonGroup.defaultProps = {
  setActive: null,
  labels: [],
  activeIndex: null,
}

export default ButtonGroup
