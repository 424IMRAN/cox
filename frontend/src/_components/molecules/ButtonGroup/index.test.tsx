import React from 'react'
import { mount, shallow } from 'enzyme'
import ButtonGroup from '.'

describe('ButtonGroup Component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <ButtonGroup labels={['test', 'bar']} activeIndex={0} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Allows us to set props', () => {
    const wrapper = mount(
      <ButtonGroup labels={['test', 'bar']} activeIndex={0} />
    )
    expect(wrapper.props().labels).toEqual(['test', 'bar'])
    expect(wrapper.props().activeIndex).toEqual(0)
    wrapper.setProps({ labels: ['bar', 'foo'] })
    wrapper.setProps({ activeIndex: 1 })
    expect(wrapper.props().activeIndex).toEqual(1)
    expect(wrapper.props().labels).toEqual(['bar', 'foo'])
  })

  it('onClick works', () => {
    const mockOnClick = jest.fn()
    const component = mount(
      <ButtonGroup
        labels={['test', 'bar']}
        activeIndex={1}
        setActive={mockOnClick}
      />
    )
    component
      ?.find('div')
      ?.at(1)
      ?.props()
      ?.onClick?.({} as any)
    expect(mockOnClick).toHaveBeenCalledWith(0)
  })
})
