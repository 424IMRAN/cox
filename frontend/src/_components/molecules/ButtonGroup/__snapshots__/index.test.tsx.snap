// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Component Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-typographyDiv-4 makeStyles-activeTypoGraphyDiv-3"
    key="test"
    onClick={[Function]}
  >
    <Typography
      align="left"
      className="makeStyles-typogrpahy-2 makeStyles-activeTypoGraphy-5"
      variant="body2"
    >
      test
    </Typography>
  </div>
  <div
    className="makeStyles-typographyDiv-4"
    key="bar"
    onClick={[Function]}
  >
    <Typography
      align="left"
      className="makeStyles-typogrpahy-2"
      variant="body2"
    >
      bar
    </Typography>
  </div>
</div>
`;
